service/src/main/java/uk/gov/hmcts/reform/fpl/service/RecordFinalDecisionsService.java

```diff
@@ -81,7 +81,8 @@ public List<Element<Child>> updateChildren(CaseData caseData) {
         List<ChildFinalDecisionDetails> childFinalDecisionDetails = eventData.getAllChildrenDecisionDetails();
 
         for (int i = 0; i < updatedChildren.size(); i++) {
-            ChildFinalDecisionReason decisionReason = childFinalDecisionDetails.get(i).getFinalDecisionReason();
+            ChildFinalDecisionReason decisionReason = childFinalDecisionDetails.get(i) != null
+                ? childFinalDecisionDetails.get(i).getFinalDecisionReason() : null;
```

Comment:
will `childFinalDecisionDetails` be null?

Reply:
No when its "null" its filled with null objects instead like this: [null, null, null, null] so doing a null check on childFinalDecisionDetails itself would evaluate to false